### DISCUSSION TITLE
Generating docs from a custom tailwind.config.js

### DISCUSSION THREAD

**Discussion by krestaino** [tailwindlabs/tailwindcss#2477] [Ideas]
It would be awesome if the docs could somehow utilize a project's custom `tailwind.config.js` and generate tailored docs with your custom values. The use case for this is working on a team, we could generate our own version of the Tailwinds docs instead of pointing new devs to the official docs, where some colors, utility names, and extended utilities would be missing.

**Comment by TobiObeck** [comment]
There is already a feature proposal (https://github.com/tailwindcss/docs/issues/81) to generate a cheat sheet based on the project which asks for basically the same, as far as I understand it.

**Comment by mtsknn** [comment]
This would be great!

**Comment by brandonpittman** [comment]


**Reply by mtsknn** [reply]
I would say yes and no.

The tool provides visual examples of many classes (but not all; lots of classes are missing), and I can see how that's useful. (In my case, I don't personally need visual examples of the classes since I have configured Tailwind to use our company's design tokens that are previewable elsewhere.)

The visual examples however also make the page very long and thus not very efficient when you just want to see a quick reference of all available classes.

**Reply by mtsknn** [reply]
The parent comment linked to https://github.com/rogden/tailwind-config-viewer. Adding it here since I can see this tool being useful for some.

**Comment by mtsknn** [comment]
Different people obviously need different solutions. Here's the style I'm thinking about (sadly not customizable nor open source): https://nerdcave.com/tailwind-cheat-sheet

This reference page shows all classes and their values, has a search functionality and links to the classes' documentation pages at https://tailwindcss.com/. The last one is handy because this way the reference remains lightweight but you can easily go to the official documentation for the textual descriptions and visual examples. (The visual examples can be useful even if you have customizations, unless you have customized _everything_.)

What would be even greater is if you could include more information in the reference than just the class names and their values. For example, I'd like to include the names of the design tokens that are configured for the classes:

| Class | Value | Design token
| -- | -- | --
| `.p-1` | `padding: 0.25rem;` (4px) | `my-tokens-spacing-1`
| `.text-xs` | `font-size: 0.75rem;` (12px) | `my-tokens-typography-sizing-xsmall`

I wonder if this could be achieved using JSDoc comments in the config file and parsing them when generating the reference...

Yeah, it might be difficult to create a one-size-fits-all tool that pleases everyone.

**Comment by paulmelero** [comment]
I'm going to drop this here, it could be useful for someone 😉. It uses [Style Dictionary](https://amzn.github.io/style-dictionary/#/README) under the hood.

- 👉🏽 https://dev.to/philw_/using-style-dictionary-to-transform-tailwind-config-into-scss-variables-css-custom-properties-and-javascript-via-design-tokens-24h5
